shared/src/main/scala/async/Timer.scala

```diff
@@ -0,0 +1,71 @@
+package gears.async
+
+import Future.Promise
+import AsyncOperations.sleep
+
+import java.util.concurrent.TimeoutException
+import scala.collection.mutable
+import scala.concurrent.TimeoutException
+import scala.util.{Failure, Success, Try}
+import gears.async.Listener
+import scala.concurrent.duration._
+import scala.annotation.tailrec
+import java.util.concurrent.CancellationException
+
+/** 
+ * Timer exposes a steady Async.Source of ticks that happens every `tickDuration` milliseconds.
+ * Note that the timer does not start ticking until `start` is called (which is a blocking operation, until the timer is cancelled).
+ * 
+ * You might want to manually `cancel` the timer, so that it gets garbage collected (before the enclosing `Async` scope ends).
+*/
+class Timer(tickDuration: Duration) extends Cancellable {
+  var isCancelled = false
+
+  private class Source extends Async.OriginalSource[this.TimerEvent] {
+    def tick() = synchronized {
+      listeners.foreach(_.completeNow(TimerEvent.Tick, this))
```

Comment:
This violates the source-attached-listener contract as the listener will be completed multiple times.